### PR TITLE
(Bug Fix)Added Name Validations for required

### PR DESCRIPTION
**Pull request by praneethreddy-4**
Added name validations in required fields

Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is point to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for Existing Functionality
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments

### CODE REVIEW

File: Ginger/Ginger/SolutionWindows/TargetApplicationsPage.xaml.cs [hunk 1]
```diff
@@ -120,34 +125,44 @@ private void SaveHandler(object sender, RoutedEventArgs e)
...
-            {
-                ApplicationPlatform currentApp = (ApplicationPlatform)xTargetApplicationsGrid.CurrentItem;
+            if (e.Column.DisplayIndex ==1)//App Name Column
+            {                
+                ApplicationPlatform currentApp = (ApplicationPlatform)xTargetApplicationsGrid.CurrentItem;                               
                 mSolution.SetUniqueApplicationName(currentApp);
 
                 if (currentApp.AppName != currentApp.NameBeforeEdit)
                 {
+                    if (currentApp.AppName == null || string.IsNullOrEmpty(currentApp.AppName.ToString()) || string.IsNullOrWhiteSpace(currentApp.AppName.ToString()))
```
Comment (Maheshkale447) [inline Ginger/Ginger/SolutionWindows/TargetApplicationsPage.xaml.cs:151]: String is null or empty will do the null check, no need to explicitly check for null

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/SolutionWindows/TargetApplicationsPage.xaml.cs (hunk 1 above)
Comment (praneethreddy-4) [reply Ginger/Ginger/SolutionWindows/TargetApplicationsPage.xaml.cs]: Removed extra condition and time complexity issue is handled.
Hence the comments stated are resolved

**Comment by Maheshkale447** [comment]
Please add unit test cases for checking duplicate names for entities
